### PR TITLE
Testing Tenants with KES Integration on PRs

### PR DESCRIPTION
**Pull request by dvaldivia**
Signed-off-by: Daniel Valdivia <18384552+dvaldivia@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/kes.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      # Runs a set of commands using the runners shell
+      - name: Deploy a MinIO Tenant with KES
+        run: |
+          curl -sLO "https://dl.k8s.io/release/v1.23.1/bin/linux/amd64/kubectl" -o kubectl
+          chmod +x kubectl
+          mv kubectl /usr/local/bin
+          
+          curl -sLO "https://dl.min.io/client/mc/release/linux-amd64/mc" -o mc
+          chmod +x mc
+          mv mc /usr/local/bin
```
Comment (harshavardhana) [inline .github/workflows/kes.yml:38]: can you not do this with `minio/mc` container?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: testing/common.sh [hunk 2]
```diff
@@ -21,3 +21,65 @@ die() {
...
+	if [ "$totalwait" -gt 305 ]; then
+	    echo "Unable to create tenant after 5 minutes, exiting."
+	    try false
+	fi
+    done
+
+    echo "Waiting for pods to be ready. (5m timeout)"
+
+    MINIO_ACCESS_KEY=$(kubectl -n $1 get secrets $2-env-configuration -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_USER="' | sed -e 's/export MINIO_ROOT_USER="//g')
+    MINIO_SECRET_KEY=$(kubectl -n $1 get secrets $2-env-configuration -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_PASSWORD="' | sed -e 's/export MINIO_ROOT_PASSWORD="//g')
```
Comment (harshavardhana) [inline testing/common.sh:73]:
```suggestion

    USER=$(kubectl -n $1 get secrets $2-env-configuration -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_USER="' | sed -e 's/export MINIO_ROOT_USER="//g')
    PASSWORD=$(kubectl -n $1 get secrets $2-env-configuration -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_PASSWORD="' | sed -e 's/export MINIO_ROOT_PASSWORD="//g')
```

File: testing/common.sh [hunk 3]
```diff
@@ -21,3 +21,65 @@ die() {
...
+
+    try kubectl wait --namespace $1 \
+        --for=condition=ready pod \
+        --selector=v1.min.io/tenant=$2 \
+        --timeout=300s
+
+    echo "Tenant is created successfully, proceeding to validate 'mc admin info minio/'"
+
+    kubectl run admin-mc -i --tty --image minio/mc --command -- bash -c "until (mc alias set minio/ https://minio.$1.svc.cluster.local $MINIO_ACCESS_KEY $MINIO_SECRET_KEY); do echo \"...waiting... for 5secs\" && sleep 5; done; mc admin info minio/;"
```
Comment (harshavardhana) [inline testing/common.sh:82]:
```suggestion
    kubectl run admin-mc -i --tty --image minio/mc --command -- bash -c "until (mc alias set minio/ https://minio.$1.svc.cluster.local $USER $PASSWORD); do echo \"...waiting... for 5secs\" && sleep 5; done; mc admin info minio/;"
```

File: testing/test-kes.sh [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  check_tenant_status default kes-tenant
+
+  echo "Port Forwarding tenant"
+  try kubectl port-forward $(kubectl get pods -l v1.min.io/tenant=kes-tenant | grep -v NAME | awk '{print $1}' | head -1) 9000 &
+
+  MINIO_ACCESS_KEY=$(kubectl -n default get secrets kes-tenant-env-configuration -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_USER="' | sed -e 's/export MINIO_ROOT_USER="//g')
+  MINIO_ROOT_PASSWORD=$(kubectl -n default get secrets kes-tenant-env-configuration -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_PASSWORD="' | sed -e 's/export MINIO_ROOT_PASSWORD="//g')
+
+  try mc config host add kestest https://localhost:9000 $MINIO_ACCESS_KEY $MINIO_ROOT_PASSWORD --insecure
+  try mc admin kms key status kestest --insecure
```
Comment (harshavardhana) [inline testing/test-kes.sh:120]:
```suggestion
  USER=$(kubectl -n default get secrets kes-tenant-env-configuration -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_USER="' | sed -e 's/export MINIO_ROOT_USER="//g')
  PASSWORD=$(kubectl -n default get secrets kes-tenant-env-configuration -o go-template='{{index .data "config.env"|base64decode }}' | grep 'export MINIO_ROOT_PASSWORD="' | sed -e 's/export MINIO_ROOT_PASSWORD="//g')

  try mc config host add kestest https://localhost:9000 $USER $PASSWORD --insecure
  try mc admin kms key status kestest --insecure
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: testing/test-kes.sh [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+
+  echo $CREDENTIALS
+
+  check_tenant_status default kes-tenant
```
Comment (harshavardhana) [inline testing/test-kes.sh:111]: This namespace is wrong I think @dvaldivia

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: testing/test-kes.sh (hunk 5 above)
Comment (dvaldivia) [reply testing/test-kes.sh:111]: no, this test uses the tenant in `default` :P

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by Alevsk — APPROVED** [review]

**Pull request merged**